src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1327,9 +1351,19 @@ static int client_receive_message(
 }
 
 int sd_dhcp6_client_stop(sd_dhcp6_client *client) {
+        int r;
+
         if (!client)
                 return 0;
 
+        if (client->send_release) {
+                client_set_state(client, DHCP6_STATE_STOPPING);
+                r = dhcp6_client_send_message(client);
+                if (r < 0)
+                        return log_dhcp6_client_errno(client, r,
+                                                      "Failed to send to DHCP6 release message: %m");
+        }
```

Comment:
Not a big matter, but I'd like to extract the logic into a separate function, e.g. `client_send_release()` or so.